inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/extensions/DynamicAnnotatorWorkflowActionBarItemGroup.java

```diff
@@ -0,0 +1,132 @@
+
+
+
+public class DynamicAnnotatorWorkflowActionBarItemGroup extends Panel
+{
+
+    private static final long serialVersionUID = -292514874000914541L;
+
+    private final DocumentService documentService;
+    private final ProjectService projectService;
+
+    private final AnnotationPageBase page;
+    protected final ConfirmationDialog finishDocumentDialog;
+    private final LambdaAjaxLink finishDocumentLink;
+    private final AnnotationQueueOverviewDataProvider provider;
+
+    @Autowired
+    public DynamicAnnotatorWorkflowActionBarItemGroup(
+        String aId, AnnotationPageBase aPage, DocumentService aDocumentService,
+        ProjectService aProjectService)
+    {
+        super(aId);
+
+        //Same as for the default
+        page = aPage;
+        projectService = aProjectService;
+        documentService = aDocumentService;
+
+        provider = new AnnotationQueueOverviewDataProvider(
+            documentService.listAnnotationDocuments(aPage.getModelObject().getProject()),
+            documentService.listSourceDocuments(
+                aPage.getModelObject().getProject()), documentService);
+
+        add(finishDocumentDialog = new ConfirmationDialog("finishDocumentDialog",
+            new StringResourceModel("FinishDocumentDialog.title", this, null),
+            new StringResourceModel("FinishDocumentDialog.text", this, null)));
+
+        add(finishDocumentLink = new LambdaAjaxLink("showFinishDocumentDialog",
+            this::actionFinishDocument));
+        finishDocumentLink.setOutputMarkupId(true);
+        finishDocumentLink.add(enabledWhen(() -> page.isEditable()));
+        finishDocumentLink.add(new Label("state")
+            .add(new CssClassNameModifier(LambdaModel.of(this::getStateClass))));
+    }
+
+    protected AnnotationPageBase getAnnotationPage()
+    {
+        return page;
+    }
+
+    public String getStateClass()
+    {
+        return FontAwesome5IconType.check_circle_r.cssClassName();
+    }
+
+    protected void actionFinishDocument(AjaxRequestTarget aTarget)
+    {
+        finishDocumentDialog.setConfirmAction((_target) ->
+        {
+            page.actionValidateDocument(_target, page.getEditorCas());
+
+            AnnotatorState state = page.getModelObject();
+            AnnotationDocument annotationDocument = documentService
+                .getAnnotationDocument(state.getDocument(), state.getUser());
+
+            documentService.transitionAnnotationDocumentState(annotationDocument,
+                ANNOTATION_IN_PROGRESS_TO_ANNOTATION_FINISHED);
+
+            documentService.createAnnotationDocument(annotationDocument);
+
+
+            //Get a new random document from the list and open it
+            SourceDocument doc = provider.getRandomDocument
+                (getAnnotationPage(), annotationDocument);
+            if (doc == null)
+            {
+                info("No more documents to annotate available. Please contact your project manager.");
```

Comment:
Note that info / error etc. messages are only shown if the feedback panel is added to the AJAX target.